### ISSUE TITLE
Support for libraries and standard model snippets in Umple

### ISSUE THREAD

**Issue by Nava2** [umple/umple#224]
Originally reported on Google Code with ID 325
Owned by @vahdat-ab

---

Part 1. Build some standard pieces of model that could be imported into other models.
For example, a standard core of a transportation system with legs, bookings, scheduled
and actual etc. 

Part 2. Add to the 'use' clause a capability that would allow renaming of certain elements
from the reused components, so for example Trip could become Flight. There may be other
ways of doing this other than adjusting the use clause.

Part 3. Allow any Umple model to declare it will use any of the built-in standard models.
In other words, adjust the search path to do the following. First look in the local
directory, then look in a standard place in the compiler release, and finally look
on the web in a standard place. In addition, it should be possible to specify a URL
in the use clause to refer to a model snippet on the web.

See also issue 324 since some existing repositories might have good examples for a
standard model library.

The RoutesAndLocations example could be part of a standard.

See also the work on VML done by Jenya Levin 

---

Reported by @umple on 2012-08-16 17:21:56

**Comment by Nava2** [comment]
See also issue 239

---

Reported by @umple on 2012-09-30 10:30:56

**Comment by Nava2** [comment]
Reported by @umple on 2012-12-31 18:08:50
- **Status changed**: `Accepted`

**Comment by Nava2** [comment]
This is part of the broader work on separation of concerns

---

Reported by @umple on 2013-07-08 15:13:31
- **Owner changed**: @vahdat-ab

**Comment by Nava2** [comment]
Reported by @umple on 2013-09-17 13:12:30
- **Labels added**: Type-ResearchGrad
- **Labels removed**: Type-ProjectUG

**Comment by TimLethbridge** [comment]
This has been started. There is an umpleonline/umplibrary/reusable director. All files here are always available to any umple program by a simple use statement, since they are made available in the jar
